NDTensors/src/lib/Sectors/src/category_product.jl

```diff
@@ -8,22 +11,30 @@ CategoryProduct(c::CategoryProduct) = _CategoryProduct(categories(c))
 
 categories(s::CategoryProduct) = s.cats
 
-Base.isempty(S::CategoryProduct) = isempty(categories(S))
-Base.length(S::CategoryProduct) = length(categories(S))
-Base.getindex(S::CategoryProduct, args...) = getindex(categories(S), args...)
+# =================================  Sectors interface  ====================================
+function SymmetryStyle(c::CategoryProduct)
+  return reduce(combine_styles, map(SymmetryStyle, categories(c)); init=EmptyCategory())
+end
+
+function quantum_dimension(::NonAbelianGroup, s::CategoryProduct)
+  return prod(map(quantum_dimension, categories(s)))
+end
 
-function fusion_rule(s1::CategoryProduct, s2::CategoryProduct)
-  return [
-    CategoryProduct(l) for l in categories_fusion_rule(categories(s1), categories(s2))
-  ]
+function quantum_dimension(::NonGroupCategory, s::CategoryProduct)
+  return prod(map(quantum_dimension, categories(s)))
 end
 
+GradedAxes.dual(s::CategoryProduct) = CategoryProduct(map(GradedAxes.dual, categories(s)))
```

Comment:
```suggestion
GradedAxes.dual(s::CategoryProduct) = CategoryProduct(GradedAxes.dual.(categories(s)))
```
Just a style thing, I like using broadcast when the calls are simple enough since it is a bit more compact than calling `map`.

Reply:
Here the `map` is on purpose to support both `Tuple` and `NamedTuple` implementation, whereas broadcasting fails on `NamedTuple`.

Reply:
Gotcha, that makes sense, strange that broadcast isn't defined for `NamedTuple`.